lib/rspec/support/spec/encoded_string_matcher.rb

```diff
@@ -0,0 +1,38 @@
+require 'rspec/matchers'
+# Special matcher for comparing encoded strings so that
+# we don't run any expectation failures through the Differ,
+# which also relies on EncodedString. Instead, confirm the
+# strings have the same encoding and same bytes.
+RSpec::Matchers.define :be_identical_string do |expected|
+
+  if String.method_defined?(:encoding)
+    match do
+      actual.encoding == expected_encoding &&
+        actual.bytes.to_a == expected.bytes.to_a
+    end
+
+    failure_message do
+      "expected #{actual.inspect} (#{actual.encoding.name}) to be identical to "\
+        "#{expected.inspect} (#{expected_encoding.name})"
+    end
+  else
+    match do |actual|
+      actual.split(//) == expected.split(//)
+    end
+  end
+
+  chain :with_encoding do |enc|
+    @expected_encoding = enc
+  end
+
+  # HACK: To disable the encoding check
+  chain :without_encoding do
+    @without_encoding = :without_encoding
+  end
```

Comment:
re: the hack here and below, I'm wondering if I should change the name of the matcher to `expect_identical_encoded_string` and `expect_identical_string` and see if there are still any cases for chaining